### DISCUSSION TITLE
Is it possible to specify the source IP for backchannel requests?

### DISCUSSION THREAD

**Discussion by studersi** [OpenIDC/mod_auth_openidc#1283] [Q&A] [answered]
There are a number of cases where mod_auth_openidc creates requests to other services. Examples would be the userinfo or instrospection lookups.

Is it possible to specify a source IP address for these requests?

Modules like mod_proxy implement this with the [ProxySourceAddress](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#proxysourceaddress) directive.

Also, libcurl implements this with [CURLOPT_INTERFACE ](https://curl.se/libcurl/c/CURLOPT_INTERFACE.html):

> This sets the interface name to use as outgoing network interface. The name can be an interface name, an **IP address**, or a hostname.

**Comment by zandbelt** [comment]
right now this is not possible but it shouldn't take much to add it: we'll look into it

**Comment by zandbelt** [comment, marked as answer]
see 8fb9cea101db052e280b1e9ed4746e75f7a6533b, which will be part of the future releases >= 2.4.16.6

**Reply by studersi** [reply]
Perfect, thank you!

**Comment by zandbelt** [comment]
https://github.com/OpenIDC/mod_auth_openidc/releases/tag/v2.4.16.6
